projects/angular/src/button/button-group/button-group.ts

```diff
@@ -136,4 +166,14 @@ export class ClrButtonGroup implements AfterContentInit {
 
     this.popoverPosition = (ClrPopoverPositions as Record<string, any>)[this._menuPosition];
   }
+
+  private handleFocusOnMenuOpen() {
+    if (this.menuButtons.length) {
+      this.toggleService.openChange.pipe(delay(0), takeUntil(this.destroy$)).subscribe(isOpened => {
+        this.focusHandler.menuToggle = this.menuToggle?.nativeElement;
+        this.focusHandler.menu = this.menu?.nativeElement;
+        this.focusHandler.menuOpened = isOpened;
```

Comment:
This `focusHandler.menuOpened` setter has a side effect, and this code does some implicitly-required setup by setting the `menu` and `menuToggle` elements.

I think a better way would be to explicitly call the `focusHandler.initialize` method if the menu is open and pass the elements. This makes the code more explicit about what is happening and what is required.

The `focusHandler.menuOpened` property is not needed at all after this change.

```typescript
if (isOpened) {
  this.focusHandler.initialize({
    menu: this.menu.nativeElement,
    menuToggle: this.menuToggle.nativeElement,
  });
}
```